torch/csrc/TypeInfo.cpp

```diff
@@ -159,11 +159,8 @@ static PyObject* THPIInfo_min(THPIInfo* self, void*) {
 }
 
 static PyObject* THPIInfo_dtype(THPIInfo* self, void*) {
-  std::string primary_name, legacy_name;
-  std::tie(primary_name, legacy_name) = torch::utils::getDtypeNames(self->type);
-  return AT_DISPATCH_INTEGRAL_TYPES(self->type, "dtype", [primary_name] {
```

Comment:
The old implementation of `THPIInfo_dtype` does not handle quantized data types:

```
>>> ii = torch.iinfo(th.quint8)
>>> ii
libc++abi.dylib: terminating with uncaught exception of type c10::Error: "dtype" not implemented for 'QUInt8'
Exception raised from operator() at ../torch/csrc/TypeInfo.cpp:166 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) + 98 (0x111401162 in libc10.dylib)
frame #1: THPIInfo_dtype(THPIInfo*, void*) + 845 (0x1107cc9fd in libtorch_python.dylib)
frame #2: THPIInfo_str(THPIInfo*) + 467 (0x1107cbff3 in libtorch_python.dylib)
...
```

The new implementation in this PR should fix the problem.

Reply:
great! 😄